public/app/features/explore/state/query.ts

```diff
@@ -570,61 +582,69 @@ export const runQueries = (
-        // checking the type of enabled supp queries
-        // then for which enabled supp queries has data source support
-        // and then we need to run the supp queries
-        // but we need to make sure that supp queries that dont work
-        // return undefined provider
-        // we should also make sure we store the type of provider that
-        // was last stored
-      } else if (hasLogsVolumeSupport(datasourceInstance)) {
-        // we always prepare the logsVolumeProvider,
-        // but we only load it, if the logs-volume-histogram is enabled.
-        // (we need to have the logsVolumeProvider always actual,
-        // even when the visuals are disabled, because when the user
-        // enables the visuals again, we need to load the histogram,
-        // so we need the provider)
-        const sourceRequest = {
-          ...transaction.request,
-          requestId: transaction.request.requestId + '_log_volume',
-        };
-        const type = SupplementaryQueryType.LogsVolume;
-        const dataProvider = datasourceInstance.getLogsVolumeDataProvider(sourceRequest);
-        dispatch(
-          storeSupplementaryQueryDataProviderAction({
-            exploreId,
-            type,
-            dataProvider,
-          })
-        );
-
-        const { supplementaryQueries, absoluteRange } = getState().explore[exploreId]!;
-        if (!canReuseSupplementaryQueryData(supplementaryQueries[type].data, queries, absoluteRange)) {
-          dispatch(cleanSupplementaryQueryAction({ exploreId, type }));
-          if (supplementaryQueries[type].enabled) {
-            dispatch(loadSupplementaryQueryData(exploreId, type));
+      } else {
+        for (const type of supplementaryQueryTypes) {
+          // We always prepare provider, even is supplementary query is disabled because when the user
+          // enables the query, we need to load the data, so we need the provider
+          if (hasSupplementaryQuerySupport(datasourceInstance, type)) {
+            const dataProvider = datasourceInstance.getDataProvider(type, {
+              ...transaction.request,
+              requestId: `${transaction.request.requestId}_${snakeCase(type)}`,
+            });
+            dispatch(
+              storeSupplementaryQueryDataProviderAction({
+                exploreId,
+                type,
+                dataProvider,
+              })
+            );
+
+            if (!canReuseSupplementaryQueryData(supplementaryQueries[type].data, queries, absoluteRange)) {
+              dispatch(cleanSupplementaryQueryAction({ exploreId, type }));
+              if (supplementaryQueries[type].enabled) {
+                dispatch(loadSupplementaryQueryData(exploreId, type));
+              }
+            }
+            // Code below (else if scenario) is for backward compatibility with data sources that don't support supplementary queries
+            // TODO: Remove after v10 (#61298)
```

Comment:
Did you mean?

```suggestion
            // TODO: Remove after v10 (#61296)
```

Reply:
I wanted to link this PR as reference to what should be removed.

Reply:
But I can create issue for us - log squad and link that

Reply:
Changed in https://github.com/grafana/grafana/pull/61298/commits/578d9b0f20abd68d9e698bd8a709f9d8716ca5b6 and new issue is here https://github.com/grafana/grafana/issues/61845